### PR TITLE
fix: fix editor bug

### PR DESCRIPTION
**Pull request by wumaolinmaoan**
## Summary by CodeRabbit

- **New Features**
	- Enhanced scene version handling, compatibility adjustments, and item ID transformations in asset migrations.
	- Added support for tracking original item IDs (`oldId`) in composition components for testing purposes.
	- Introduced a new method to validate specific object types in serialization processes.
	- Implemented improved time tracking between frames in the ticker functionality.
- **Refactor**
	- Updated composition and player components for better id management and playback handling.
	- Adjusted import paths in the spine component for better module organization.
- **Bug Fixes**
	- Fixed issues with item end behavior and time handling in compositions.

### CODE REVIEW

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/composition.ts [hunk 1]
```diff
@@ -106,15 +105,15 @@ export class Composition implements Disposable, LostHandler {
   /**
    * 单个合成结束时的回调
    */
-  onEnd?: (composition: Composition) => void;
+  onEnd: (composition: Composition) => void;
```
[the remark this answers is not in the document]
Comment (wumaolinmaoan) [reply packages/effects-core/src/composition.ts]: 之前是创建的时候给默认值了所以这边问号去了，我加回来吧

File: packages/effects-core/src/composition.ts [hunk 2]
```diff
@@ -285,7 +284,9 @@ export class Composition implements Disposable, LostHandler {
     this.buildItemTree(this.rootItem);
     this.rootItem.onEnd = () => {
       window.setTimeout(() => {
-        this.handleEnd?.(this);
+        if (this.onEnd) {
```
Comment (yiiqii) [inline packages/effects-core/src/composition.ts:287]: this.onEnd?.(this);

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/composition.ts (hunk 2 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/composition.ts:287]: 已修改

Comment (yiiqii) [reply packages/effects-core/src/composition.ts:287]: > 已修改

代码没提上来？

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/composition.ts (hunk 2 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/composition.ts]: 已提交

**Review by yiiqii — APPROVED** [review]

**Pull request merged**